ParkingZoneApp/Areas/Admin/Controllers/ParkingZonesController.cs

```diff
@@ -62,17 +65,19 @@ public IActionResult Edit(Guid? id)
             if (parkingZone is null)
                 return NotFound();
 
-            _parkingZoneService.Update(parkingZone);
-            return View(parkingZone);
+            var parkingZoneVM = new EditVM(parkingZone);
+            return View(parkingZoneVM);
         }
 
         // POST: Admin/ParkingZones/Edit/5
         // To protect from overposting attacks, enable the specific properties you want to bind to.
         // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public IActionResult Edit(Guid id, ParkingZone parkingZone)
+        public IActionResult Edit(Guid id, EditVM parkingZoneEditVM)
         {
+            var parkingZone = new EditVM().MapToModel(parkingZoneEditVM);
```

Comment:
```suggestion
            var parkingZone = parkingZoneEditVM.(parkingZone);
```

Reply:
before doing that find with the service the parkingZone if exists or not and then give to the parameter